File: CPU_scheduling/menu.c
#include <stdio.h>
#include <stdlib.h>

int main() {

    int choice;

    while(1) {

        printf("\nCPU Scheduling Simulator\n");
        printf("------------------------\n");
        printf("1. Enter Process Data\n");
        printf("2. FCFS\n");
        printf("3. SJF\n");
        printf("4. SRTF\n");
        printf("5. Round Robin\n");
        printf("6. Priority (Preemptive)\n");
        printf("7. Priority (Non-Preemptive)\n");
        printf("8. Compare Algorithms\n");
        printf("9. Exit\n");

        printf("\nSelect Option: ");
        scanf("%d", &choice);

        switch(choice) {

            case 1:
                system("./input ");
                break;

            case 2:
                system("./fcfs ");
                break;

            case 3:
                system("./sjf ");
                break;

            case 4:
                system("./srtf ");
                break;

            case 5:
                system("./rr ");
                break;

            case 6:
                system("./priority ");
                break;

            case 7:
                system("./priority_np ");
                break;

            case 8: {

                FILE *fp = fopen("comparison_data.txt","w");
                if(fp != NULL)
                    fclose(fp);

                system("./fcfs ");
                system("./sjf ");
                system("./srtf ");
                system("./rr ");
                system("./priority ");
                system("./priority_np ");
                

                system("python3 compare.py  ");

                break;
            }

            case 9:
                exit(0);

            default:
                printf("Invalid choice\n");
        }
    }

    return 0;
}

File: CPU_scheduling/menu.py
import os

def run_program(source, exe):
    
    os.system(f"gcc {source} -o {exe} -lm")

    os.system(f"./{exe}")


while True:

    print("\n===================================")
    print("      CPU Scheduling Simulator")
    print("===================================")
    print("1. Input Processes")
    print("2. FCFS Scheduling")
    print("3. SJF Scheduling")
    print("4. SRTF Scheduling")
    print("5. Round Robin Scheduling")
    print("6. Priority Scheduling (Preemptive)")
    print("7. Priority Scheduling (Non-Preemptive)")
    print("8. Compare All Algorithms")
    print("0. Exit")
    print("===================================")

    choice = input("Enter your choice: ")

    if choice == "1":
        run_program("input.c", "input")

    elif choice == "2":
        run_program("fcfs.c", "fcfs")

    elif choice == "3":
        run_program("sjf.c", "sjf")

    elif choice == "4":
        run_program("srtf.c", "srtf")

    elif choice == "5":
        run_program("rr.c", "rr")

    elif choice == "6":
        run_program("priority.c", "priority")

    elif choice == "7":
        run_program("priority_np.c", "priority_np")

    elif choice == "8":
        print("\nRunning All Algorithms for Comparison...\n")

        run_program("fcfs.c", "fcfs")
        run_program("sjf.c", "sjf")
        run_program("srtf.c", "srtf")
        run_program("rr.c", "rr")
        run_program("priority.c", "priority")
        run_program("priority_np.c", "priority_np")

    elif choice == "0":
        print("Exiting Program...")
        break

    else:
        print("Invalid choice! Try again.")